lib/matplotlib/dates.py

```diff
@@ -201,6 +220,65 @@ def _get_rc_timezone():
     MO, TU, WE, TH, FR, SA, SU)
 WEEKDAYS = (MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY, SATURDAY, SUNDAY)
 
+# default epoch: passed to np.datetime64...
+_epoch = None
+_epoch_used = False
+
+
+def _reset_epoch():
+    """
+    Reset the Matplotlib date epoch to its rcParams value.
+    """
+    global _epoch
+    global _epoch_used
+
+    _epoch_used = False
+    _epoch = matplotlib.rcParams['date.epoch']
+
+
+def set_epoch(epoch):
+    """
+    Set the epoch (origin for dates) for datetime calculations.  The
+    default epoch is '1970-01-01T00:00:00'. The `.date2num` method returns
```

Comment:
I'd rephrase this paragraph to:

For most of the date-related operations you won't need to adapt the epoch. Exceptions are when you are relying on the internal float representation of dates in matplotlib or in some cases when you microsecond resolution.

The default epoch is :rc:`dates.epoch`. Since matplotlib internally represents dates as days since the epoch, the resolution is limited to ~20µs for dates further than 70 years from the epoch.

Reply:
Fair enough. How about 

```
    The default epoch is :rc:`dates.epoch`. Since matplotlib internally
    represents dates as days since the epoch, the resolution is by floating
    point round off that depends on the number of days since the epoch.  If
    microsecond accuracy is desired, the date being plotted needs to be within
    70 years of the epoch.
```

Reply:
Missing word? the resolution is *given* by ...

I want to convey two things:
- First block) Most people will never need to set the epoch. They should not worry about the epoch at all.
- Second block) Quantify when this gets relevant and how much the precision loss is.

Reply:
```
    The default epoch is :rc:`dates.epoch`, and users should not need to
    change this.

    If microsecond accuracy is desired, the date being plotted needs to be
    within approximately 70 years of the epoch. Matplotlib internally
    represents dates as days since the epoch, so floating point dynamic
    range needs to be within a factor fo 2^52.

    `~dates.set_epoch` must be called before any dates are converted
    (i.e. near the import section) or a RuntimeError will be raised.
```

Reply:
Sounds good.